composites/Plugin/SnippetEditor/text.js

```diff
@@ -0,0 +1,114 @@
+import { EditorState, SelectionState, Modifier } from "draft-js";
+
+/**
+ * Test if a character in a text is whitespace. Out of range index counts as whitespace.
+ *
+ * @param {string} text  The text to get the character from.
+ * @param {int}    index The index of the character in the text.
+ *
+ * @returns {boolean} True if the character is out of bounds or whitespace.
+ */
+export const hasWhitespaceAt = ( text, index ) => {
+	const char = text.charAt( index );
+
+	// No character found equals whitespace.
+	if ( char.length === 0 ) {
+		return true;
+	}
+
+	return /\s/.test( char );
+};
+
+/**
+ * Gets the Draft.js caret offset.
+ *
+ * @param {SelectionState} selection The Draft.js selection state.
+ *
+ * @returns {int} The Draft.js caret offset.
+ */
+export const getCaretOffset = ( selection ) => {
+	const isBackward = selection.getIsBackward();
+
+	// If the selection is backwards the end offset is the caret.
+	return isBackward ? selection.getEndOffset() : selection.getStartOffset();
+};
+
+/**
+ * Gets the block of the start of the current Draft.js selection.
+ *
+ * This is a tiny helper function to get the anchor key and then the block of that key.
+ * Read more about the Draft.js selection state: https://draftjs.org/docs/api-reference-selection-state.html
+ * Read more about the Draft.js content block: https://draftjs.org/docs/api-reference-content-block.html
+ *
+ * @param {ContentState}   content   The Draft.js content state.
+ * @param {SelectionState} selection The Draft.js selection state.
+ *
+ * @returns {ContentBlock} The Draft.js ContentBlock.
+ */
+export const getAnchorBlock = ( content, selection ) => {
+	const key = selection.getAnchorKey();
+	return content.getBlockForKey( key );
+};
+
+/**
+ * Inserts text into the editor state.
+ *
+ * @param {EditorState} editorState The Draft.js editor state.
+ * @param {string}      text        The text to insert.
+ *
+ * @returns {EditorState} The new editor state.
+ */
+export const insertText = ( editorState, text ) => {
+	const content = editorState.getCurrentContent();
+	const selection = editorState.getSelection();
+
+	// If there is a selection instead of a caret do nothing.
```

Comment:
please add comma after 'caret' to improve readability

Reply:
Also: this, in combination with the function name sounds like you can't insert text when you have something selected. Is that really the case?